lib/openid_connect/discovery/provider/config/response.rb

```diff
@@ -83,6 +83,12 @@ def jwks
             JSON::JWK::Set.new @jwks[:keys]
           end
 
+          def public_keys_hash
+            @public_keys_hash ||= jwks.collect do |jwk|
+              {:kid => jwk[:kid], :key => JSON::JWK.decode(jwk)}
```

Comment:
I guess it would be nicer to have kid as member of the key but at the moment the key is a OpenSSL::PKey:: instance. Maybe there is some easy way of doing that? other solution would be to create a new key class that wraps OpenSSL::PKey::